remapping_pipeline.nf

```diff
@@ -0,0 +1,350 @@
+    awk '{print $1"\\t"$2-1"\\t"$2"\\t"$3"\\t"$4"\\t"$5"\\t"$6"\\t"$7}' variants_remapped.vcf > variants_remapped.bed
+    
+    # Getfasta to get the REF genome alleles and then extract the genome alleles without the positions
+    bedtools getfasta -tab -fi genome.fa -bed variants_remapped.bed | awk '{print $2}'> genome_alleles.txt
+    
+    # Make all alleles caps (sometimes there are lower-case bases)
+    tr a-z A-Z < genome_alleles.txt > genome_alleles.fixed.txt
+    
+    # Paste all the information into the correct columns
+    cut -f1-3 variants_remapped.vcf > temp_first_3_columns.txt
+    cut -f4-7 variants_remapped.vcf > temp_last_4_columns.txt
+    paste temp_first_3_columns.txt genome_alleles.fixed.txt temp_last_4_columns.txt > var_pre_final.vcf
+    '''
+}
+
+
+/*
+ * 
+ */
+process buildHeader {
+
+    input:
+        path "var_pre_final.vcf" from var_pre_final
+        path "vcf_header.txt" from vcf_header
+
+    output:
+        path "final_header.txt" into final_header
+
+    """
+    # Create list of contigs/chromosomes to be added to the header
+    cut -f 1 var_pre_final.vcf | sort -u > contig_names.txt
+    while read CHR; do echo "##contig=<ID=\${CHR}>"; done < contig_names.txt > contigs.txt
+    # Add the reference assembly
+    echo "##reference=${params.newgenome}" >> contigs.txt
+    
+    # Copy everything that isn't #CHROM (the column titles), ##contig or ##reference from the old header to a temp
+    awk '(\$1 !~ /^##contig/ && \$1 !~ /^##reference/ && \$1 !~ /^#CHROM/) {print \$0}' vcf_header.txt > temp_header.txt
+
+    # Add the two headers together and add the column names
+    cat temp_header.txt contigs.txt > final_header.txt
+    echo -e "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO" >> final_header.txt
+    """
+}
+
+
+/*
+ * 
+ */
+process mergeHeaderAndContentAndFixRefAllele {
+
+    input:
+        path "final_header.txt" from final_header
+        path "var_pre_final.vcf" from var_pre_final
+        path "old_ref_alleles.txt" from old_ref_alleles
+
+
+    output:
+        path "pre_final_vcf.vcf" into pre_final_vcf
+        
+    """
+    # Add header to the vcf file:
+    cat final_header.txt var_pre_final.vcf > vcf_out_with_header.vcf
+    # Also add it to the ref alleles file so that the line numbers match:
+    cat final_header.txt old_ref_alleles.txt > old_ref_alleles_with_header.txt
+    # Test each variant to see if REF = ALT, if so, replace the current REF with the corresponding old REF (this is to 
+    # deal with variants such as G > G)
+    ${baseDir}/replace_refs.py -i vcf_out_with_header.vcf -r old_ref_alleles_with_header.txt -o pre_final_vcf.vcf
+    """
+}
+
+/*
+ * 
+ */
+process normalise {
+
+    publishDir outfile_dir,
+        overwrite: true,
+        mode: "copy"
+
+    input:
+        path "genome.fa" from params.newgenome
+        path "pre_final_vcf.vcf" from pre_final_vcf
+
+    output:
+        path "${outfile_basename}" into output_file        
```

Comment:
`output_file` is very non-descriptive, it should be renamed to something like `output_vcf` or even `final_output_vcf`